### ISSUE TITLE
Pagination Issue - Pagination URL is null if pagination node key contains a `.`

### ISSUE THREAD

**Issue by leighgibbo** [craftcms/feed-me#1384]
### Description

TL;DR: In the settings for "Pagination URL", selecting a node with a dot in it, results in a null value for `paginationUrl` being returned for each respective job, and pagination does not work.

Selecting a pagination URL node key that contains a period `.` saves the feed config as expected 🙂, but prevents the actual jobs from fetching the correct URL 🙁, as [the `HASH::get`](https://github.com/craftcms/feed-me/blob/develop/src/base/DataType.php#L49) method will [split the supplied key into arrays to allow for some simple dot-notation parsing](https://github.com/cakephp/cakephp/blob/5.x/src/Utility/Hash.php#L62).

### Steps to reproduce

1.  Connect to a feed that returns nodes with periods/dots in the keys, such as an oData API (returns things like `@odata.nextLink`, or `@odata.count`)
2. Select a key with a period contained
3. Run the job and check the Job's Data in Queue Manager - the `paginationUrl` will be null

### Additional info

- Craft version: 4.5.5
- PHP version: 8.1.21
- Database driver & version: MariaDB-1:10.4.30
- Plugins & versions: FeedMe 5.0.4
